Missing logs when cloudstack-setup-agent is run with sudo

### problem

Trying to setup a AlmaLinux 9.5 KVM host with Cloudstack 4.20.

So far, I followed the instructions in the wiki and created a cloudstack user with sudo permissions.
When running `sudo -u cloudstack cloudstack-setup-agent`, it did spit some strange error message
```
Please input the Hypervisor type kvm/lxc:[kvm]
DEBUG:root:execute:route -n|awk '/^0.0.0.0/ {print $2,$8}'
Failed to get default route. Please configure your network to have a default route

```

When running `route -n|awk '/^0.0.0.0/ {print $2,$8}'` alone it worked well.
So I started hacking into the python code, and added a raise statement.

This time, I got the following error message
```
PermissionError: [Errno 13] Permission denied: '/bin/sh'
```

This allowed me to debug and find out that I had  `Defaults noexec` in my `/etc/sudoers` file set, even when running sudo I wasn't allowed to run `/bin/sh`.

The problem here is that running `cloudstack-setup-agent` with sudo doesn't create any error log in   `/var/log/cloudstack/agent/setup.log` file, even if running `sudo -u cloudstack echo "Test" >> /var/log/cloudstack/agent/setup.log` works.

As a side note, the bash class should perhaps send those exception errors to stderr too


### versions

The versions of ACS, hypervisors, storage, network etc..

### The steps to reproduce the bug

1.
2.
3.
...


### What to do about it?

_No response_

I double checked that the labels are set correctly. All the bridges are up and connected to ethernet interfaces.
When I add the host on UI, I get the same error message as in the screenshot above (530: Failed to setup keystore on the KVM host).
On the KVM host, journalctl -r shows the following entries (no errors):

```
avril 15 12:00:01 redacted_kvm_host.local sudo[54135]:     root : no tty ; PWD=/root ; USER=root ; COMMAND=/usr/share/cloudstack-common/scripts/util/keystore-setup /etc/cloudstack/agent/agent.properties /etc/cloudstack/agent/cloud.jks hrguMbgNwpX4bw8w 365 /etc/cloudstack/agent/cloud.csr
```

On the management server, `/var/log/cloudstack/management/management-server.log` relevant output
```
2025-04-15 12:00:01,054 INFO  [c.c.u.e.CSExceptionErrorCode] (qtp1513608173-22181:[ctx-2414a1f9, ctx-8130a03c]) (logid:b90fe3f8) Could not find exception: com.cl
oud.exception.DiscoveryException in error code list for exceptions
2025-04-15 12:00:01,054 WARN  [o.a.c.a.c.a.h.AddHostCmd] (qtp1513608173-22181:[ctx-2414a1f9, ctx-8130a03c]) (logid:b90fe3f8) Exception: com.cloud.exception.Disco
veryException: Could not add host at [http://redacted_host_name.local] with zone [1], pod [1] and cluster [1] due to: [ can't setup agent, due to com.cloud.utils.e
xception.CloudRuntimeException: Failed to setup keystore on the KVM host: 10.13.37.2 - Failed to setup keystore on the KVM host: 10.13.37.2].
        at com.cloud.resource.ResourceManagerImpl.discoverHostsFull(ResourceManagerImpl.java:835)
        at com.cloud.resource.ResourceManagerImpl.discoverHosts(ResourceManagerImpl.java:661)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:569)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:344)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:198)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:215)
        at jdk.proxy3/jdk.proxy3.$Proxy223.discoverHosts(Unknown Source)
        at org.apache.cloudstack.api.command.admin.host.AddHostCmd.execute(AddHostCmd.java:134)
        at com.cloud.api.ApiDispatcher.dispatch(ApiDispatcher.java:173)
        at com.cloud.api.ApiServer.queueCommand(ApiServer.java:831)
        at com.cloud.api.ApiServer.handleRequest(ApiServer.java:652)
        at com.cloud.api.ApiServlet.processRequestInContext(ApiServlet.java:354)
        at com.cloud.api.ApiServlet$1.run(ApiServlet.java:157)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:56)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:103)
        at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:53)
        at com.cloud.api.ApiServlet.processRequest(ApiServlet.java:154)
        at com.cloud.api.ApiServlet.doPost(ApiServlet.java:113)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:665)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
        at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1450)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:554)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:505)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:772)
        at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
        at org.eclipse.jetty.server.Server.handle(Server.java:516)
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487)
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
        at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: com.cloud.exception.DiscoveredWithErrorException:  can't setup agent, due to com.cloud.utils.exception.CloudRuntimeException: Failed to setup keystore
 on the KVM host: 10.13.37.2 - Failed to setup keystore on the KVM host: 10.13.37.2
        at com.cloud.hypervisor.kvm.discoverer.LibvirtServerDiscoverer.find(LibvirtServerDiscoverer.java:379)
        at com.cloud.resource.ResourceManagerImpl.discoverHostsFull(ResourceManagerImpl.java:828)
        ... 60 more
Caused by: com.cloud.utils.exception.CloudRuntimeException: Failed to setup keystore on the KVM host: 10.13.37.2
        at com.cloud.hypervisor.kvm.discoverer.LibvirtServerDiscoverer.setupAgentSecurity(LibvirtServerDiscoverer.java:181)
        at com.cloud.hypervisor.kvm.discoverer.LibvirtServerDiscoverer.find(LibvirtServerDiscoverer.java:324)
        ... 61 more
```

Of course, keytool is installed.
I decided to manually run the given keystore setup command on the KVM host:
```
 export LANG=C
[root@host root]# /usr/share/cloudstack-common/scripts/util/keystore-setup /etc/cloudstack/agent/agent.properties /etc/cloudstack/agent/cloud.jks hrguMbgNwpX4bw8w 365 /etc/cloudstack/agent/cloud.csr

Generating 2,048 bit RSA key pair and self-signed certificate (SHA256withRSA) with a validity of 365 days
        for: CN=hyper02p.val.npf.local, OU=cloudstack, O=cloudstack, C=cloudstack
-----BEGIN NEW CERTIFICATE REQUEST-----
[redacted]
-----END NEW CERTIFICATE REQUEST-----
```
So running the keystore script looks good.
Even after this step, trying to add the KVM host via the UI failed with the same 530 error.

--- a/scripts/util/keystore-setup
+++ b/scripts/util/keystore-setup
@@ -22,48 +22,71 @@ KS_PASS="$3"
 KS_VALIDITY="$4"
 CSR_FILE="$5"
 
+LOG_FILE="/var/log/cloudstack/agent/keystore-setup.log"
+
 ALIAS="cloud"
 LIBVIRTD_FILE="/etc/libvirt/libvirtd.conf"
 
+log() {
+    __log_line="${1}"
+    __log_level="${2:-INFO}"
+
+    __log_line="${__log_level}: ${__log_line}"
+    echo "${__log_line}" >> "${LOG_FILE}"
+    echo "${__log_line}"
+}
+
+log "$(date) - starting keystore-setup" >> "${LOG_FILE}"
+
 # Re-use existing password or use the one provided
 if [ -f "$PROPS_FILE" ]; then
-    OLD_PASS=$(sed -n '/keystore.passphrase/p' "$PROPS_FILE" 2>/dev/null  | sed 's/keystore.passphrase=//g' 2>/dev/null)
+    log "Previous props file exists, trying to extract password"
+    OLD_PASS=$(sed -n '/^keystore.passphrase/p' "$PROPS_FILE" 2>> "${LOG_FILE}"  | sed 's/^keystore.passphrase=//g' 2>> "${LOG_FILE}")
     if [ ! -z "${OLD_PASS// }" ]; then
         KS_PASS="$OLD_PASS"
+        log "Password extraction successful"
     else
-        sed -i "/keystore.passphrase.*/d" $PROPS_FILE 2> /dev/null || true
+        sed -i "/^keystore.passphrase.*/d" $PROPS_FILE 2>> "${LOG_FILE}" || true
         echo "keystore.passphrase=$KS_PASS" >> $PROPS_FILE
+        if [ $? != 0 ]; then
+                log "Could not add new password to agent.properties" "ERROR"
+        else
+                log "New keystore password set"
+        fi
     fi
 fi
 
 if [ -f "$KS_FILE" ]; then
-    keytool -delete -noprompt -alias "$ALIAS" -keystore "$KS_FILE" -storepass "$KS_PASS" > /dev/null 2>&1 || true
+    log "keystore file exists. Deleting current entries"
+    keytool -delete -noprompt -alias "$ALIAS" -keystore "$KS_FILE" -storepass "$KS_PASS" >> "${LOG_FILE}" || log "Failed to delete current entries" "ERROR"
 fi
 
+log "Generating new key"
 CN=$(hostname --fqdn)
-keytool -genkey -storepass "$KS_PASS" -keypass "$KS_PASS" -alias "$ALIAS" -keyalg RSA -validity "$KS_VALIDITY" -dname cn="$CN",ou="cloudstack",o="cloudstack",c="cloudstack" -keystore "$KS_FILE" > /dev/null 2>&1
+keytool -genkey -storepass "$KS_PASS" -keypass "$KS_PASS" -alias "$ALIAS" -keyalg RSA -validity "$KS_VALIDITY" -dname cn="$CN",ou="cloudstack",o="cloudstack",c="cloudstack" -keystore "$KS_FILE" >> "${LOG_FILE}" 2>&1
 
 # Generate CSR
-rm -f "$CSR_FILE"
+log "Generating CSR"
+[ -f "$CSR_FILE" ] && rm -f "$CSR_FILE"
 addresses=$(ip address | grep inet | awk '{print $2}' | sed 's/\/.*//g' | grep -v '^169.254.' | grep -v '^127.0.0.1' | egrep -v '^::1|^fe80' | grep -v '^::1' | sed 's/^/ip:/g' | tr '\r\n' ',')
-keytool -certreq -storepass "$KS_PASS" -alias "$ALIAS" -file $CSR_FILE -keystore "$KS_FILE" -ext san="$addresses" > /dev/null 2>&1
-
+log "Found following SAN addresses to add to CSR: ${addresses}"
+keytool -certreq -storepass "$KS_PASS" -alias "$ALIAS" -file $CSR_FILE -keystore "$KS_FILE" -ext san="$addresses" >> "${LOG_FILE}" 2>&1
 if [ $? -ne 0 ];then
-    echo "Failed to generate CSR file, retrying after removing existing settings"
+    log "Failed to generate CSR file, retrying after removing existing settings" "ERROR"
 
     if [ -f "$LIBVIRTD_FILE" ]; then
-        echo "Reverting libvirtd to not listen on TLS"
+        log "Reverting libvirtd to not listen on TLS"
         sed -i "s,^listen_tls=1,listen_tls=0,g" $LIBVIRTD_FILE
         systemctl restart libvirtd
     fi
 
-    echo "Removing cloud.* files in /etc/cloudstack/agent"
-    rm -f /etc/cloudstack/agent/cloud.*
+    log "Removing cloud.* files in /etc/cloudstack/agent"
+    rm -f /etc/cloudstack/agent/cloud.* || log "Could not remove /etc/cloudstack/agent/cloud.*" "ERROR"
 
-    echo "Retrying to generate CSR file"
-    keytool -certreq -storepass "$KS_PASS" -alias "$ALIAS" -file $CSR_FILE -keystore "$KS_FILE" -ext san="$addresses" >/dev/null 2>&1
+    log "Retrying to generate CSR file"
+    keytool -certreq -storepass "$KS_PASS" -alias "$ALIAS" -file $CSR_FILE -keystore "$KS_FILE" -ext san="$addresses" >> "${LOG_FILE}" 2>&1
     if [ $? -ne 0 ];then
-        echo "Failed to generate CSR file while retrying"
+        log "Failed to generate CSR file while retrying" "ERROR"
         exit 1
     fi
 fi
@@ -71,6 +94,6 @@ fi
 cat "$CSR_FILE"
 
 # Fix file permissions
-chmod 600 $KS_FILE
-chmod 600 $PROPS_FILE
-chmod 600 $CSR_FILE
+chmod 600 $KS_FILE || log "Cannot chmod $KS_FILE" "ERROR"
+chmod 600 $PROPS_FILE || log "Cannot chmod $PROPS_FILE" "ERROR"
+chmod 600 $CSR_FILE || log "Cannot chmod $CSR_FILE" "ERROR"

--- a/scripts/util/keystore-setup
+++ b/scripts/util/keystore-setup
@@ -22,71 +22,78 @@ KS_PASS="$3"
 KS_VALIDITY="$4"
 CSR_FILE="$5"
 
-LOG_FILE="/var/log/cloudstack/agent/keystore-setup.log"
-
 ALIAS="cloud"
 LIBVIRTD_FILE="/etc/libvirt/libvirtd.conf"
 
-log() {
-    __log_line="${1}"
-    __log_level="${2:-INFO}"
+if type -p logger > /dev/null; then
+    LOGGER_CMD="$(type -p logger) -t cloudstack-keystore-setup"
+else
+    LOG_FILE="/var/log/cloudstack/agent/cloudstack-keystore-setup.log"
+    log() {
+        if [ "${1}" != "" ]; then
+            __log_line="${1}"
+        else
+            read -r __log_line
+        fi
 
-    __log_line="${__log_level}: ${__log_line}"
-    echo "${__log_line}" >> "${LOG_FILE}"
-    echo "${__log_line}"
-}
+        echo "${__log_line}" >> "${LOG_FILE}"
+        echo "${__log_line}"
+    }
+    LOGGER_CMD=log
+fi
 
-log "$(date) - starting keystore-setup" >> "${LOG_FILE}"
+$LOGGER_CMD "$(date) - starting keystore-setup"
 
 # Re-use existing password or use the one provided
 if [ -f "$PROPS_FILE" ]; then
-    log "Previous props file exists, trying to extract password"
-    OLD_PASS=$(sed -n '/^keystore.passphrase/p' "$PROPS_FILE" 2>> "${LOG_FILE}"  | sed 's/^keystore.passphrase=//g' 2>> "${LOG_FILE}")
-    if [ ! -z "${OLD_PASS// }" ]; then
+    $LOGGER_CMD "Previous props file exists, trying to extract password"
+    OLD_PASS=$(sed -n '/^keystore.passphrase/p' "$PROPS_FILE" | sed 's/^keystore.passphrase=//g')
+    if [ -n "${OLD_PASS// }" ]; then
         KS_PASS="$OLD_PASS"
-        log "Password extraction successful"
+        $LOGGER_CMD "Password extraction successful"
     else
-        sed -i "/^keystore.passphrase.*/d" $PROPS_FILE 2>> "${LOG_FILE}" || true
-        echo "keystore.passphrase=$KS_PASS" >> $PROPS_FILE
+        sed -i "/^keystore.passphrase.*/d" "$PROPS_FILE" 2>&1 | $LOGGER_CMD || true
+        echo "keystore.passphrase=$KS_PASS" >> "$PROPS_FILE"
         if [ $? != 0 ]; then
-                log "Could not add new password to agent.properties" "ERROR"
+                $LOGGER_CMD "Could not add new password to agent.properties"
         else
-                log "New keystore password set"
+                $LOGGER_CMD "New keystore password set"
         fi
     fi
 fi
 
 if [ -f "$KS_FILE" ]; then
-    log "keystore file exists. Deleting current entries"
-    keytool -delete -noprompt -alias "$ALIAS" -keystore "$KS_FILE" -storepass "$KS_PASS" >> "${LOG_FILE}" || log "Failed to delete current entries" "ERROR"
+    $LOGGER_CMD "keystore file exists. Deleting current entries"
+    keytool -delete -noprompt -alias "$ALIAS" -keystore "$KS_FILE" -storepass "$KS_PASS" 2>&1 | $LOGGER_CMD
+    [ $? -ne 0 ] && $LOGGER_CMD "Failed to delete current entries"
 fi
 
-log "Generating new key"
+$LOGGER_CMD "Generating new key"
 CN=$(hostname --fqdn)
-keytool -genkey -storepass "$KS_PASS" -keypass "$KS_PASS" -alias "$ALIAS" -keyalg RSA -validity "$KS_VALIDITY" -dname cn="$CN",ou="cloudstack",o="cloudstack",c="cloudstack" -keystore "$KS_FILE" >> "${LOG_FILE}" 2>&1
+keytool -genkey -storepass "$KS_PASS" -keypass "$KS_PASS" -alias "$ALIAS" -keyalg RSA -validity "$KS_VALIDITY" -dname cn="$CN",ou="cloudstack",o="cloudstack",c="cloudstack" -keystore "$KS_FILE" 2>&1 | $LOGGER_CMD
 
 # Generate CSR
-log "Generating CSR"
+$LOGGER_CMD "Generating CSR"
 [ -f "$CSR_FILE" ] && rm -f "$CSR_FILE"
 addresses=$(ip address | grep inet | awk '{print $2}' | sed 's/\/.*//g' | grep -v '^169.254.' | grep -v '^127.0.0.1' | egrep -v '^::1|^fe80' | grep -v '^::1' | sed 's/^/ip:/g' | tr '\r\n' ',')
-log "Found following SAN addresses to add to CSR: ${addresses}"
-keytool -certreq -storepass "$KS_PASS" -alias "$ALIAS" -file $CSR_FILE -keystore "$KS_FILE" -ext san="$addresses" >> "${LOG_FILE}" 2>&1
+$LOGGER_CMD "Found following SAN addresses to add to CSR: ${addresses}"
+keytool -certreq -storepass "$KS_PASS" -alias "$ALIAS" -file "$CSR_FILE" -keystore "$KS_FILE" -ext san="$addresses" 2>&1 | $LOGGER_CMD
 if [ $? -ne 0 ];then
-    log "Failed to generate CSR file, retrying after removing existing settings" "ERROR"
+    $LOGGER_CMD "Failed to generate CSR file, retrying after removing existing settings"
 
     if [ -f "$LIBVIRTD_FILE" ]; then
-        log "Reverting libvirtd to not listen on TLS"
+        $LOGGER_CMD "Reverting libvirtd to not listen on TLS"
         sed -i "s,^listen_tls=1,listen_tls=0,g" $LIBVIRTD_FILE
         systemctl restart libvirtd
     fi
 
-    log "Removing cloud.* files in /etc/cloudstack/agent"
-    rm -f /etc/cloudstack/agent/cloud.* || log "Could not remove /etc/cloudstack/agent/cloud.*" "ERROR"
+    $LOGGER_CMD "Removing cloud.* files in /etc/cloudstack/agent"
+    rm -f /etc/cloudstack/agent/cloud.* || $LOGGER_CMD "Could not remove /etc/cloudstack/agent/cloud.*"
 
-    log "Retrying to generate CSR file"
-    keytool -certreq -storepass "$KS_PASS" -alias "$ALIAS" -file $CSR_FILE -keystore "$KS_FILE" -ext san="$addresses" >> "${LOG_FILE}" 2>&1
+    $LOGGER_CMD "Retrying to generate CSR file"
+    keytool -certreq -storepass "$KS_PASS" -alias "$ALIAS" -file "$CSR_FILE" -keystore "$KS_FILE" -ext san="$addresses" 2>&1 | $LOGGER_CMD
     if [ $? -ne 0 ];then
-        log "Failed to generate CSR file while retrying" "ERROR"
+        $LOGGER_CMD "Failed to generate CSR file while retrying"
         exit 1
     fi
 fi
@@ -94,6 +101,6 @@ fi
 cat "$CSR_FILE"
 
 # Fix file permissions
-chmod 600 $KS_FILE || log "Cannot chmod $KS_FILE" "ERROR"
-chmod 600 $PROPS_FILE || log "Cannot chmod $PROPS_FILE" "ERROR"
-chmod 600 $CSR_FILE || log "Cannot chmod $CSR_FILE" "ERROR"
+chmod 600 "$KS_FILE" || $LOGGER_CMD "Cannot chmod $KS_FILE"
+chmod 600 "$PROPS_FILE" || $LOGGER_CMD "Cannot chmod $PROPS_FILE"
+chmod 600 "$CSR_FILE" || $LOGGER_CMD "Cannot chmod $CSR_FILE"

